accounts-db/src/accounts_db.rs

```diff
@@ -1777,112 +1777,7 @@ impl AccountsDb {
-            ),
-            uncleaned_pubkeys: DashMap::new(),
-            next_id: AtomicAccountsFileId::new(0),
-            shrink_candidate_slots: Mutex::new(ShrinkCandidates::default()),
-            write_cache_limit_bytes: None,
-            write_version: AtomicU64::new(0),
-            paths: vec![],
-            base_working_path,
-            base_working_temp_dir,
-            accounts_hash_cache_path,
-            shrink_paths: Vec::default(),
-            temp_paths: None,
-            file_size: DEFAULT_FILE_SIZE,
-            thread_pool: rayon::ThreadPoolBuilder::new()
-                .num_threads(num_threads)
-                .thread_name(|i| format!("solAccounts{i:02}"))
-                .stack_size(ACCOUNTS_STACK_SIZE)
-                .build()
-                .unwrap(),
-            thread_pool_clean: make_min_priority_thread_pool(),
-            thread_pool_hash: make_hash_thread_pool(),
-            bank_hash_stats: Mutex::new(bank_hash_stats),
-            accounts_delta_hashes: Mutex::new(HashMap::new()),
-            accounts_hashes: Mutex::new(HashMap::new()),
-            incremental_accounts_hashes: Mutex::new(HashMap::new()),
-            external_purge_slots_stats: PurgeStats::default(),
-            clean_accounts_stats: CleanAccountsStats::default(),
-            shrink_stats: ShrinkStats::default(),
-            shrink_ancient_stats: ShrinkAncientStats::default(),
-            stats: AccountsStats::default(),
-            account_indexes: AccountSecondaryIndexes::default(),
-            #[cfg(test)]
-            load_delay: u64::default(),
-            #[cfg(test)]
-            load_limit: AtomicU64::default(),
-            is_bank_drop_callback_enabled: AtomicBool::default(),
-            remove_unrooted_slots_synchronization: RemoveUnrootedSlotsSynchronization::default(),
-            shrink_ratio: AccountShrinkThreshold::default(),
-            dirty_stores: DashMap::default(),
-            zero_lamport_accounts_to_purge_after_full_snapshot: DashSet::default(),
-            accounts_update_notifier: None,
-            log_dead_slots: AtomicBool::new(true),
-            exhaustively_verify_refcounts: false,
-            accounts_file_provider: AccountsFileProvider::default(),
-            storage_access: StorageAccess::default(),
-            scan_filter_for_shrinking: ScanFilter::default(),
-            partitioned_epoch_rewards_config: PartitionedEpochRewardsConfig::default(),
-            epoch_accounts_hash_manager: EpochAccountsHashManager::new_invalid(),
-            test_skip_rewrites_but_include_in_bank_hash: false,
-            latest_full_snapshot_slot: SeqLock::new(None),
-            is_experimental_accumulator_hash_enabled: default_accounts_db_config
-                .enable_experimental_accumulator_hash
-                .into(),
-            best_ancient_slots_to_shrink: RwLock::default(),
-        }
+        Self::new_single_for_tests()
```

Comment:
I believe this substitution in `default_for_tests()` is valid; here are relevant functions that are getting called now:
https://github.com/anza-xyz/agave/blob/0e16e60e08fcfdb1b4a12f19ca4be4054a2707dc/accounts-db/src/accounts_db.rs#L1888-L1914

A few changes as far as I can tell:
- In the old code, `default_with_accounts_index()` would leave `.paths` and `.temp_paths` fields empty, whereas new code will create the temp dirs and corresponding paths + populate them
- The previous code used `AccountsDbConfig::default()` / defaults for fields whereas new code will use `ACCOUNTS_DB_CONFIG_FOR_TESTING`. This is still test code so I think this is fine too

I might lack some context / be missing something, but I'm currently under the impression that `AccountsDb::default_for_tests()` is redundant and can be removed. That seems like something for a subsequent PR tho

Reply:
yeah. we can remove `default_for_tests().